lib/babe/verify_test.go

```diff
@@ -5,396 +5,1227 @@ package babe
 
 import (
 	"errors"
+	"fmt"
 	"math/big"
 	"testing"
-	"time"
 
-	"github.com/ChainSafe/gossamer/dot/state"
 	"github.com/ChainSafe/gossamer/dot/types"
-	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
-	"github.com/ChainSafe/gossamer/lib/genesis"
 	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func newTestVerificationManager(t *testing.T, genCfg *types.BabeConfiguration) *VerificationManager {
-	testDatadirPath := t.TempDir()
-
-	config := state.Config{
-		Path:     testDatadirPath,
-		LogLevel: log.Info,
+func newTestHeader(t *testing.T, digest ...scale.VaryingDataTypeValue) *types.Header {
+	t.Helper()
+	header := types.NewEmptyHeader()
+	for _, d := range digest {
+		err := header.Digest.Add(d)
+		assert.NoError(t, err)
 	}
-	dbSrv := state.NewService(config)
-	dbSrv.UseMemDB()
 
-	if genCfg == nil {
-		genCfg = genesisBABEConfig
-	}
+	return header
+}
 
-	gen, genTrie, genHeader := genesis.NewDevGenesisWithTrieAndHeader(t)
-	err := dbSrv.Initialise(gen, genHeader, genTrie)
+func signAndAddSeal(t *testing.T, kp *sr25519.Keypair, header *types.Header, data []byte) {
+	t.Helper()
+	sig, err := kp.Sign(data)
 	require.NoError(t, err)
 
-	err = dbSrv.Start()
-	require.NoError(t, err)
-	dbSrv.Epoch, err = state.NewEpochStateFromGenesis(dbSrv.DB(), dbSrv.Block, genCfg)
-	require.NoError(t, err)
-
-	logger.Patch(log.SetLevel(defaultTestLogLvl))
+	err = header.Digest.Add(types.SealDigest{
+		ConsensusEngineID: types.BabeEngineID,
+		Data:              sig,
+	})
+	assert.NoError(t, err)
+}
 
-	vm, err := NewVerificationManager(dbSrv.Block, dbSrv.Epoch)
-	require.NoError(t, err)
-	return vm
+func newEncodedBabeDigest(t *testing.T, value scale.VaryingDataTypeValue) ([]byte, error) {
+	t.Helper()
+	babeDigest := types.NewBabeDigest()
+	err := babeDigest.Set(value)
+	assert.NoError(t, err)
+	return scale.Marshal(babeDigest)
 }
 
-func TestVerificationManager_OnDisabled_InvalidIndex(t *testing.T) {
-	vm := newTestVerificationManager(t, nil)
+func encodeAndHashHeader(t *testing.T, header *types.Header) (common.Hash, error) {
```

Comment:
If you're going to keep this a test helper function, I don't think it should return an error.  As mentioned in @qdm12, you should `require.NoError` in these cases, and not return an error from this function.

On the flip side we we don't pass `*testing.T` to this function at all and just `require.NoError` on the calling side.

Reply:
errors have been removed from helpers now and asserts are now requires